src/planwise/endpoint/projects2.clj

```diff
@@ -31,7 +31,8 @@
    (PUT "/:id" [id project :as request]
      (let [user-id  (util/request-user-id request)
            id       (Integer. id)
-           project  (assoc project :id id)] ;; honor id of route
+           project  (-> project (assoc :id id)
+                        (dissoc :dataset-sites))] ;; honor id of route
```

Comment:
This shouldn't be necessary. Eventually, the projects component should filter out invalid/unknown attributes. Extra attributes shouldn't matter for the spec either.